esy/NpmRelease.ml

```diff
@@ -302,73 +158,71 @@ let make ~esyInstallRelease ~outputPath ~concurrency ~cfg ~sandbox =
+        let synPkg = {
+          Package.
+          id = "__release_env__";
+          name = "release-env";
+          version = pkg.version;
+          dependencies = [Package.Dependency pkg];
+          sourceType = Manifest.SourceType.Development;
+          sandboxEnv = pkg.sandboxEnv;
+          build = Package.EsyBuild {
               buildCommands = None;
               installCommands = None;
               buildType = Manifest.BuildType.OutOfSource;
             };
-            exportedEnv = [];
-            sourcePath = pkg.sourcePath;
-            resolution = None;
-          } in
-          let%bind task = Task.ofPackage
+          exportedEnv = [];
+          sourcePath = pkg.sourcePath;
+          resolution = None;
+        } in
+        let%bind task = Task.ofPackage
             ~term:(Some "$TERM")
             ~forceImmutable:true
             ~overrideShell:false
             synPkg
-          in
-          return (Environment.Closed.bindings task.Task.env)
         in
-
-        Environment.renderToShellSource
-          ~localStorePath:(Path.v "/local/store/does/not/exist")
-          ~storePath:(Path.v "$ESY__STORE")
-          ~sandboxPath:(Path.v "$ESY__RELEASE")
-        env
+        return (Environment.Closed.value task.Task.env)
       ) in
-      let%bind () = Fs.writeFile ~data Path.(outputPath / "releaseEnv") in
-      return ()
+
+    let binPath = Path.(outputPath / "bin") in
+    let%bind () = Fs.createDir binPath in
+
+    (* Emit wrappers for released binaries *)
+    let%bind () =
+      let%bind env = RunAsync.ofRun (Environment.Value.bindToConfig cfg env) in
+
+      let generateBinaryWrapper name =
+        let resolveBinInEnv ~env prg =
+          let path =
+            let v = match StringMap.find_opt "PATH" env with
+              | Some v  -> v
+              | None -> ""
+            in
+            String.split_on_char ':' v
+          in RunAsync.ofRun (Run.ofBosError (Cmd.resolveCmd path prg))
+        in
+        let%bind namePath = resolveBinInEnv ~env name in
+        (* Create the .ml file that we will later compile and write it to disk *)
+        let data = makeBinWrapper ~environment:env ~bin:namePath in
+        let mlPath = Path.(binPath / (name ^ ".ml")) in
+        let%bind () = Fs.writeFile ~data mlPath in
+        (* Compile the wrapper to a binary *)
+        let compile = Cmd.(v (p ocamlopt) % "-o" % p Path.(binPath / name) % "unix.cmxa" % p mlPath) in
+        let%bind () = ChildProcess.run compile in
+        (* Replace the storePath with a string of equal length containing only _ *)
+        let%bind origPrefix, destPrefix =
+          let nextStorePrefix = String.make (String.length (Path.toString cfg.storePath)) '_' in
+          return (cfg.storePath, Path.v nextStorePrefix)
+        in
+        let%bind () = Fs.writeFile ~data:(Path.to_string destPrefix) Path.(binPath / "_storePath") in
+        Task.rewritePrefix ~cfg ~origPrefix ~destPrefix binPath
```

Comment:
I think `Task.rewritePrefix` rewrites all files in a directory and we could it do it once after we compile all executables.